software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+         put (pGUI["nodes"][tEnd]["horz"]-tClean) & "," & pGUI["nodes"][tEnd]["vert"] into tEndCoord
+         put tLink into tBox["link"]
+         put tStartCoord & cr & tEndCoord into tBox["line"]
+         put pSizes["link"] into tBox["size"]
+         put pInfo["links"][tLink]["type"] into tBox["type"]
+         put pInfo["links"][tLink]["weight"] into tWeight
+         if tWeight is empty then
+            put 1 into pGUI["links"][tLink]["weight"]
+         else
+            put tWeight into pGUI["links"][tLink]["weight"]
+         end if
+         put tBox["type"] into pGUI["links"][tLink]["type"]
+         put makeLink(tBox) into pGUI["links"][tLink]["graphic"]
+      else if pInfo["links"][tLInk]["type"] is "dive" then -- DIVE 
+         -- what node diving into
+         --- pInfo link>read=#
+         --- pGUI node>#>graphic
+         -- rect of that control
+         -- points = L-10,top cr L,T+10
+         put pInfo["links"][tLink]["read"] into tTheNode
+         put pGUI["nodes"][tTheNode]["graphic"] into tTheNodeName
+         put the rect of tTheNodeName into tTheNodeRect
+         put item 1 of tTheNodeRect into tL
+         put item 2 of tTheNodeRect into tT
+         put (tL-10) & "," & tT & cr & tL & "," & (tT+10) into tStuff["line"]
+         put tLink into tStuff["link"]
+         put 2 into tStuff["size"]
+         put makeScopeLink(tStuff) into pGUI["links"][tLink]["graphic"]
+      else if pInfo["links"][tLInk]["type"] is "rise" then -- RISE
+         -- what node rising out of
+         --- pInfo link>trim=#
+         --- pGUI node>#>graphic
+         -- rect of that control 
+         -- points = right,T+10 cr R+10,top
+         put pInfo["links"][tLink]["bulk"] into tTheNode
+         put pGUI["nodes"][tTheNode]["graphic"] into tTheNodeName
+         put the rect of tTheNodeName into tTheNodeRect
+         put item 3 of tTheNodeRect into tR
+         put item 2 of tTheNodeRect into tT
+         put tR  & "," & (tT+10) & cr & (tR+10) & "," & tT into tStuff["line"]
+         put tLink into tStuff["link"]
+         put 2 into tStuff["size"]
+         put makeScopeLink(tStuff) into pGUI["links"][tLink]["graphic"]
+      end if
+   end repeat
+   -- draw scope links...also crudly -------------------
+   
+   -- make sure the node graphics are on top
+   repeat for each key tNode in pGUI["nodes"]
+      set the layer of pGUI["nodes"][tNode]["graphic"] to top
+   end repeat
+   -- then make sure the preview graphics are on top of that
+   set the layer of graphic "preFlow1" stack "HowstrBeta" to top
+   set the layer of graphic "preFlow2" stack "HowstrBeta" to top
+   set the layer of field "preState" stack "HowstrBeta" to top
+   set the layer of field "preChange" stack "HowstrBeta" to top
+   -- highlight focused node (since all nodes new, old highlighted one was deleted)
+   put pGUI["nodes"][pNow]["graphic"] into tBlerg
```

Comment:
@matthewmaier `tBlerg`? meaningful variable names can be very helpful when you come back to the code in a few months or years. We probably want to get rid of `tBlerg` and `tStuff` etc and replace with `tSelectedNode` and `tNodePropertyA`